sound/soc/sof/ipc.c

```diff
@@ -429,12 +433,16 @@ EXPORT_SYMBOL(snd_sof_ipc_msgs_rx);
 static void ipc_trace_message(struct snd_sof_dev *sdev, u32 msg_id)
 {
 	struct sof_ipc_dma_trace_posn posn;
+	int ret;
 
 	switch (msg_id) {
 	case SOF_IPC_TRACE_DMA_POSITION:
 		/* read back full message */
-		snd_sof_ipc_msg_data(sdev, NULL, &posn, sizeof(posn));
-		snd_sof_trace_update_pos(sdev, &posn);
+		ret = snd_sof_ipc_msg_data(sdev, NULL, &posn, sizeof(posn));
+		if (ret < 0)
+			dev_warn(sdev->dev, "failed to read trace position: %d\n", ret);
```

Comment:
@lyakh Can you double-check and comment this won't spam the kernel log in corner cases (Ranjani's comment). I went through related code and this seems safe as the failure would only happen at state changes, and then we get a small number of errors. Of course, if this would happen a continuous basis, this would be really problematic and warrant a ratelimited calll.

@ranj063 I'll leave this approved. You can check the response to this and go-ahead/nack the patch.

Reply:
@kv2019i @ranj063 interesting question. Are there any specific test-cases where these would be (more) prevailing? Can we check such logs in CI? I checked a couple of random logs, haven't found any.

Reply:
I think dev_warn makes more sense